astropy/units/quantity_helper/function_helpers.py

```diff
@@ -0,0 +1,648 @@
+                              np.core.overrides.ENABLE_ARRAY_FUNCTION)
+else:
+    # In 1.17, overrides are enabled by default, but it is still possible to
+    # turn them off using an environment variable.  We use getattr since it
+    # is planned to remove that possibility in later numpy versions.
+    ARRAY_FUNCTION_ENABLED = getattr(np.core.overrides,
+                                     'ENABLE_ARRAY_FUNCTION', True)
+SUBCLASS_SAFE_FUNCTIONS = set()
+"""Functions with implementations supporting subclasses like Quantity."""
+FUNCTION_HELPERS = {}
+"""Functions with implementations usable with proper unit conversion."""
+DISPATCHED_FUNCTIONS = {}
+"""Functions for which we provide our own implementation."""
+UNSUPPORTED_FUNCTIONS = set()
+"""Functions that cannot sensibly be used with quantities."""
+
+SUBCLASS_SAFE_FUNCTIONS |= {
+    np.alen, np.shape, np.size, np.ndim,
+    np.reshape, np.ravel, np.moveaxis, np.rollaxis, np.swapaxes,
+    np.transpose, np.atleast_1d, np.atleast_2d, np.atleast_3d,
+    np.expand_dims, np.squeeze, np.broadcast_to, np.broadcast_arrays,
+    np.flip, np.fliplr, np.flipud, np.rot90,
+    np.argmin, np.argmax, np.argsort, np.lexsort, np.searchsorted,
+    np.nonzero, np.argwhere, np.flatnonzero,
+    np.diag_indices_from, np.triu_indices_from, np.tril_indices_from,
+    np.real, np.imag, np.diag, np.diagonal, np.diagflat,
+    np.empty_like, np.zeros_like,
+    np.compress, np.extract, np.delete, np.trim_zeros, np.roll, np.take,
+    np.put, np.fill_diagonal, np.tile, np.repeat,
+    np.split, np.array_split, np.hsplit, np.vsplit, np.dsplit,
+    np.stack, np.column_stack, np.hstack, np.vstack, np.dstack, np.block,
+    np.amax, np.amin, np.ptp, np.sum, np.cumsum,
+    np.prod, np.product, np.cumprod, np.cumproduct,
+    np.round, np.around,
+    np.fix, np.angle, np.i0, np.clip,
+    np.isposinf, np.isneginf, np.isreal, np.iscomplex,
+    np.average, np.mean, np.std, np.var, np.median, np.trace,
+    np.nanmax, np.nanmin, np.nanargmin, np.nanargmax, np.nanmean,
+    np.nanmedian, np.nansum, np.nancumsum, np.nanstd, np.nanvar,
+    np.nanprod, np.nancumprod,
+    np.einsum_path, np.trapz, np.linspace,
+    np.sort, np.msort, np.partition, np.meshgrid,
+    np.common_type, np.result_type, np.can_cast, np.min_scalar_type,
+    np.iscomplexobj, np.isrealobj,
+    np.shares_memory, np.may_share_memory}
+
+if not NUMPY_LT_1_15:
+    SUBCLASS_SAFE_FUNCTIONS |= {np.take_along_axis, np.put_along_axis}
+
+
+# Implemented as methods on Quantity:
+# np.ediff1d is from setops, but we support it anyway; the others
+# currently return NotImplementedError.
+# TODO: move latter to UNSUPPORTED? Would raise TypeError instead.
+SUBCLASS_SAFE_FUNCTIONS |= {
+    np.ediff1d,
+    np.all, np.any, np.sometrue, np.alltrue}
+
+# Subclass safe, but possibly better if overridden (e.g., with different
+# default arguments for isclose, allclose).
+# TODO: decide on desired behaviour.
```

Comment:
Some of these TODO items should probably get their own issues?

Reply:
Fair point. See #8825.